build(aio): upgrade to @angular/*@6.0.0-rc.5

IgorMinar opened the pull request.

IgorMinar committed:
build(aio): upgrade to @angular/*@6.0.0-rc.2

IgorMinar commented:
@alexeagle the buildsize script is doing something weird. none of my changes could have caused the unrelated bundle.min.js to decrease in size.

IgorMinar committed:
build(aio): upgrade to @angular/*@6.0.0-rc.3

mary-poppins commented:
You can preview af5c7b8 at https://pr23234-af5c7b8.ngbuilds.io/.

IgorMinar committed:
build(aio): update to @angular/cli@6.0.0-rc.2 + project layout update

project layout was updated using:
yarn ng update @angular/cli --migrate-only --from=1.7.3

IgorMinar committed:
test: update cli-hello-world to cli@6.0.0-rc.2

IgorMinar committed:
test: simplify config for cli-hello-world

IgorMinar committed:
build: update to rxjs@6.0.0-tactical-rc.1

IgorMinar committed:
build(aio): update to @angular/material@5.2.4

IgorMinar committed:
build(aio): update to @angular/material@6.0.0-rc.1

alfaproject commented on aio/package.json:
```diff
@@ -156,6 +156,7 @@
     "xregexp": "^4.0.0",
-    "yargs": "^7.0.2"
+    "yargs": "^7.0.2",
+    "@angular-devkit/build-angular": "~0.5.0"
```
sort

IgorMinar replied:
fixed. thanks

IgorMinar committed:
build: fix angular.json that was missing keys due to cli bugs

https://github.com/angular/angular-cli/issues/10225
https://github.com/angular/angular-cli/issues/10226

IgorMinar committed:
build(aio): reorder entries in package.json

Commit:
docs(aio): Upgrade examples to Angular 6

gkalpak commented on integration/cli-hello-world/package.json:
```diff
@@ -26,6 +26,7 @@
     "zone.js": "file:../../node_modules/zone.js"
   },
   "devDependencies": {
+    "@angular-devkit/build-angular": "~0.5.0",
```
I suspect this should be below `@angular/...` dependencies :grin:

gkalpak committed:
build(aio): switch to `webpack-cli` for IE polyfills

gkalpak committed:
build(aio): upgrade `rxjs` to `6.0.0-turbo-rc.4`

mary-poppins commented:
You can preview 32810e4 at https://pr23234-32810e4.ngbuilds.io/.

gkalpak committed:
build(aio): remove redundant flags from cli commands

gkalpak committed:
build(aio): upgrade `@angular/*` to `6.0.0-rc.4`

gkalpak committed:
build(aio): upgrade `@angular/cli` to `6.0.0-rc.4`

gkalpak committed:
build(aio): fix deployment script

gkalpak committed:
test(aio): fix tests and update testing infra

IgorMinar committed:
build: remove a postinstall-patch to fix rxjs

IgorMinar committed:
build: update to rxjs@6.0.0-uncanny-rc.7

IgorMinar committed:
ci: chown bazel-built packages when running integration tests

mary-poppins commented:
You can preview 94d29d1 at https://pr23234-94d29d1.ngbuilds.io/.

IgorMinar committed:
build(aio): update tslint and codelyzer

mary-poppins commented:
You can preview bca3305 at https://pr23234-bca3305.ngbuilds.io/.

mary-poppins commented:
You can preview d3a7618 at https://pr23234-d3a7618.ngbuilds.io/.

IgorMinar committed:
style(aio): lint fixes for examples

mary-poppins commented:
You can preview 69e9d5a at https://pr23234-69e9d5a.ngbuilds.io/.

IgorMinar committed:
fixup! build(aio): remove redundant flags from cli commands

IgorMinar committed:
build(aio): update to angular/core@6.0.0-rc.5

mary-poppins commented:
You can preview abe0360 at https://pr23234-abe0360.ngbuilds.io/.

IgorMinar committed:
fixup! build(aio): update to angular/core@6.0.0-rc.5

IgorMinar commented:
I'm currently blocked on https://github.com/angular/angular-cli/issues/10333

IgorMinar committed:
test(aio): move reflect-metadata polyfills to test.ts

This resolves https://github.com/angular/angular-cli/issues/10333 and nicely cleans up the code.

IgorMinar commented:
@filipesilva helped me resolve angular/angular-cli#10333 - the tests now run but ~6 are failing. likely due to legit bugs introduced by this PR.

mary-poppins commented:
You can preview 42355f9 at https://pr23234-42355f9.ngbuilds.io/.

IgorMinar committed:
build(aio): update to @angular/material@6.0.0-rc.11

mary-poppins commented:
You can preview 61c2dfc at https://pr23234-61c2dfc.ngbuilds.io/.

IgorMinar committed:
build(aio): add @angular/language-service

gkalpak commented on aio/src/app/custom-elements/api/api-list.component.ts:
```diff
@@ -16,9 +16,9 @@ import { ApiSection, ApiService } from './api.service';
-  type? = 'all';
+  query ? = '';
+  status ? = 'all';
+  type ? = 'all';
```
😞

gkalpak commented on aio/tslint.json:
```diff
@@ -55,7 +58,7 @@
     "no-trailing-whitespace": true,
     "no-unused-expression": true,
-    "no-use-before-declare": true,
+    "no-use-before-declare": false,
```
Why this?

gkalpak commented on scripts/ci/test-aio.sh:
```diff
@@ -27,7 +27,7 @@ source ${thisDir}/_travis-fold.sh
 
   # Run unit tests
   travisFoldStart "test.aio.unit"
-    yarn test --single-run
```
Is this really not needed any more?

gkalpak replied:
It is indeed not needed :/

filipesilva replied:
Depends on the settings in your angular.json. The right flag is now `--watch` though.

gkalpak committed:
fixup! test(aio): move reflect-metadata polyfills to test.ts

mary-poppins commented:
You can preview 7f4db12 at https://pr23234-7f4db12.ngbuilds.io/.

gkalpak committed:
docs(aio): update `yarn test` command in README.md

gkalpak committed:
test(aio): fix failing tests

IgorMinar commented:
Ci fails because you haven't updated the lockfile

gkalpak commented:
Fixed that, let's hope it passed now :smiley:

mary-poppins commented:
You can preview b928bde at https://pr23234-b928bde.ngbuilds.io/.

mary-poppins commented:
You can preview 279d2ac at https://pr23234-279d2ac.ngbuilds.io/.

brandonroberts committed:
docs(aio): Fix failing boilerplate tests

mary-poppins commented:
You can preview 31bcb0d at https://pr23234-31bcb0d.ngbuilds.io/.

brandonroberts committed:
docs(aio): Fix failing upgrade-module tests

brandonroberts committed:
docs(aio): Bump shared dependencies to RC5

brandonroberts committed:
docs(aio): Bump shared yarn.lock file for examples

mary-poppins commented:
You can preview 557797f at https://pr23234-557797f.ngbuilds.io/.

brandonroberts commented:
@IgorMinar the changes for the example updates are included.

IgorMinar committed:
build(aio): turn on webpack's stats.json generation for debugging purposes

mary-poppins commented:
You can preview 9237e0a at https://pr23234-9237e0a.ngbuilds.io/.

IgorMinar commented on aio/content/examples/toh-pt6/e2e/src/app.e2e-spec.ts:
```diff
@@ -47,7 +47,7 @@ class Hero {
 // TODO(i): temorarily disable these tests because angular-in-memory-web-api is not compatible with rxjs v6 yet
 //          and we don't have the backwards compatibility package yet.
 //          Reenable after rxjs v6 compatibility package is out or angular-in-memory-web-api is compatible with rxjs v6
```
please remove the comment above

brandonroberts replied:
Fixed

IgorMinar commented on aio/tools/example-zipper/customizer/package-json/base.json:
```diff
@@ -18,6 +18,7 @@
     "angular-in-memory-web-api",
     "core-js",
     "rxjs",
+    "web-animations-js",
```
we shouldn't need this polyfill, remove?

brandonroberts replied:
Fixed

IgorMinar commented on aio/tools/examples/shared/boilerplate/cli/src/polyfills.ts:
```diff
@@ -50,9 +50,22 @@ import 'core-js/es7/reflect';
  * Needed for: All but Chrome, Firefox and Opera. http://caniuse.com/#feat=web-animation
  **/
-// import 'web-animations-js';
+import 'web-animations-js';  // Run `npm install --save web-animations-js`.
```
do we really needs this? it shouldn't be necessary

brandonroberts replied:
Fixed

IgorMinar commented on aio/content/examples/upgrade-module/src/systemjs.config.1.js:
```diff
@@ -27,7 +27,8 @@
       'rxjs': 'npm:rxjs',
-      'angular-in-memory-web-api': 'npm:angular-in-memory-web-api/bundles/in-memory-web-api.umd.js'
+      'angular-in-memory-web-api': 'npm:angular-in-memory-web-api/bundles/in-memory-web-api.umd.js',
+      'tslib': 'npm:tslib/tslib.js',
```
you shouldn't need tslib. I believe there was a bug in rc.4 that caused this requirement but that is now resolved, you just didn't see it because you updated to rc.5 after this change. please remove

brandonroberts replied:
Fixed

IgorMinar commented on aio/content/examples/upgrade-module/src/systemjs.config.1.js:
```diff
@@ -40,7 +41,7 @@
       'rxjs/testing': {main: 'index.js', defaultExtension: 'js' },
       'rxjs/websocket': {main: 'index.js', defaultExtension: 'js' },
-      'rxjs': { main: 'index.js', defaultExtension: 'js' },
+      'rxjs': { main: 'index.js', defaultExtension: 'js' }
```
wow. I'm shocked that I missed this. also note that ajax, operators, testing and websocket entry points are defined twice above. can you please remove the duplicates?

brandonroberts replied:
Fixed

IgorMinar reviewed (commented):
@brandonroberts I left a few comments for you above. thanks for all the changes!

gkalpak commented on aio/tools/examples/shared/boilerplate/cli/.gitignore:
```diff
@@ -35,7 +35,7 @@ testem.log
 
 # e2e
-/e2e/*.js
+/e2e/(?!protractor)*.js
```
Hm...I don't this works

brandonroberts replied:
Edit removed as we aren't copying the .gitignore file into the examples

gkalpak commented on aio/tools/examples/shared/boilerplate/cli/src/styles.css:
```diff
@@ -0,0 +1 @@
+/* You can add global styles to this file, and also import other style files */
```
Is this intentionally added?

brandonroberts replied:
It's part of the generated content from a new CLI project

brandonroberts committed:
docs(aio): Cleanup examples with edits from Igor/George

mary-poppins commented:
You can preview 9f78409 at https://pr23234-9f78409.ngbuilds.io/.

IgorMinar commented:
Thanks @brandonroberts and @gkalpak for helping to finish this big change!! I think it's ready to be merged in now.

IgorMinar commented:
@gkalpak or @brandonroberts can you please approve? I can't since it's my PR :)
